runtime/lua/vim/keymap.lua

```diff
@@ -133,4 +146,115 @@ function keymap.del(modes, lhs, opts)
   end
 end
 
+--- @class vim.keymap.get.Filter
+--- @inlinedoc
+---
+--- Lhs of mapping
+--- @field lhs? string
+---
+--- Patter to match against rhs of mapping
+--- @field rhs? string
+---
+--- Get a mapping for a certain buffer
+--- @field buffer? integer|boolean
+
+--- @class vim.keymap.get.Return.Opts
+--- @inlinedoc
+---
+--- If they mapping is an expr mapping
+--- @field expr boolean
+---
+--- If they mapping is a buffer-local mapping
+--- @field buffer boolean
+---
+--- Description of the mapping
+--- @field desc? string
+
+--- @class vim.keymap.get.Return
+--- @inlinedoc
+---
+--- Lhs of mapping
+--- @field lhs string
+---
+--- Rhs of mapping (can be callback)
+--- @field rhs string|function
+--- Mode of the mapping
+--- @field mode string
+---
+--- Get a mapping for a certain buffer
+--- @field opts? vim.keymap.get.Return.Opts
+
+--- Gets mappings in a format easily usable for vim.keymap.set
+--- Examples:
+---
+--- ```lua
+--- -- Gets all normal mode keymaps
+--- vim.keymap.get('n')
+---
+--- -- Gets a mapping which maps to a certain rhs
+--- vim.keymap.get('n', { rhs = "<Plug>(MyAmazingFunction)" })
+--- ```
+---
+---@param modes string|string[]
+---@param filter? vim.keymap.get.Filter
+---@return vim.keymap.get.Return[]
+function keymap.get(modes, filter)
+  vim.validate({
+    mode = { modes, { 's', 't' } },
+    filter = { filter, 't', true },
+  })
+
+  filter = filter or {}
+  modes = type(modes) == 'string' and { modes } or modes
+  --- @cast modes string[]
+
+  local keymaps = {}
+  for _, mode in ipairs(modes) do
+    if filter.buffer then
+      table.insert(
+        keymaps,
+        ---@diagnostic disable-next-line: param-type-mismatch
+        vim.api.nvim_buf_get_keymap(filter.buffer == true and 0 or filter.buffer, mode)
+      )
+    else
+      table.insert(keymaps, vim.api.nvim_get_keymap(mode))
+    end
+  end
+
+  local function matches(mapping)
+    local match = true
+    if filter.lhs then
+      filter.lhs = filter.lhs:gsub('<leader>', vim.g.mapleader or '')
+      filter.lhs = filter.lhs:gsub('<localleader>', vim.g.maplocaleader or '')
+      match = match and filter.lhs == mapping.lhs
+    end
+    if filter.rhs then
+      filter.rhs = filter.rhs:gsub('<leader>', vim.g.mapleader or '')
+      filter.rhs = filter.rhs:gsub('<localleader>', vim.g.maplocaleader or '')
+      match = match and string.match(mapping.rhs, filter.rhs)
```

Comment:
This won't work for `<Leader>` and `<LocalLeader>` as they have different capitalization.